Examples/src/main/kotlin/com/mapbox/vision/examples/POIView.kt

```diff
@@ -0,0 +1,167 @@
+        }
+    }
+
+    private fun preparePOIDrawData(poiStateList: List<POIState>): List<POIDrawData> {
+        val poiDrawDataList = mutableListOf<POIDrawData>()
+        for (poiState in poiStateList) {
+            val worldCoordinate = poiState.worldCoordinate ?: continue
+            val poiBitmap = poiState.poi.bitmap ?: continue
+
+            // Prepare bounding rect for POI in mobile screen coordinates
+            val poiBitmapRect = calculatePOIScreenRect(worldCoordinate)
+            val poiLabelAlpha = calculatePOILabelAlpha(poiState)
+            val poiDrawData = POIDrawData(poiBitmap, poiBitmapRect, poiLabelAlpha)
+            poiDrawDataList.add(poiDrawData)
+        }
+        return poiDrawDataList
+    }
+
+    private fun calculatePOIScreenRect(poiWorldCoordinate: WorldCoordinate): Rect {
+        // Calculate left top coordinate of POI in real world using POI world coordinate
+        val worldLeftTop = poiWorldCoordinate.copy(
+                y = poiWorldCoordinate.y + LABEL_SIZE_METERS / 2,
+                z = poiWorldCoordinate.z + LABEL_ABOVE_GROUND_METERS + LABEL_SIZE_METERS
+        )
+
+        // Calculate right bottom coordinate of POI in real world using POI world coordinate
+        val worldRightBottom = poiWorldCoordinate.copy(
+                y = poiWorldCoordinate.y - LABEL_SIZE_METERS / 2,
+                z = poiWorldCoordinate.z + LABEL_ABOVE_GROUND_METERS
+        )
+        val poiBitmapRect = Rect(0, 0, 0, 0)
+
+        // Calculate POI left top position on camera frame from real word coordinates
+        VisionReplayManager.worldToPixel(worldLeftTop)?.run {
+            poiBitmapRect.left = x.toFloat().scaleX().toInt()
+            poiBitmapRect.top = y.toFloat().scaleY().toInt()
+        }
+
+        // Calculate POI right bottom position on camera frame from real word coordinates
+        VisionReplayManager.worldToPixel(worldRightBottom)?.run {
+            poiBitmapRect.right = x.toFloat().scaleX().toInt()
+            poiBitmapRect.bottom = y.toFloat().scaleY().toInt()
+        }
+        return poiBitmapRect
+    }
+
+    private fun calculatePOILabelAlpha(poiState: POIState): Int {
+        val minDistance = min(DRAW_LABEL_MIN_DISTANCE_METERS - poiState.distanceToVehicle, DISTANCE_FOR_ALPHA_APPEAR_METERS)
+        return ((minDistance / DISTANCE_FOR_ALPHA_APPEAR_METERS.toFloat()) * 255).toInt()
+    }
+
+    override fun onDraw(canvas: Canvas) {
+        for (p in poiDrawDataList) {
+            with(p) {
+                paint.alpha = poiBitmapAlpha
+                canvas.drawBitmap(poiBitmap, null, poiBitmapRect, paint)
+            }
+        }
+        super.onDraw(canvas)
+    }
+
+    private var scaleFactor = 1f
+    private var scaledSize = ImageSize(1, 1)
+
+    // Scale position of camera frame coordinate on device screen like android:scaleType="centerCrop"
+    private fun Float.scaleX() = this * scaleFactor - (scaledSize.imageWidth - width) / 2
```

Comment:
Wasn't it easier to use `ImageView` with `centerCrop` instead of `View`?

Reply:
changed, ImageView instead of custom logic